services/iaasalpha/wait/wait.go

```diff
@@ -346,3 +349,53 @@ func DeleteVirtualIPWaitHandler(ctx context.Context, a APIClientInterface, proje
 	handler.SetTimeout(15 * time.Minute)
 	return handler
 }
+
+// ImageUploadWaitHandler will wait for server creation
+func ImageUploadWaitHandler(ctx context.Context, a APIClientInterface, projectId, imageId string) *wait.AsyncActionHandler[iaasalpha.Image] {
+	handler := wait.New(func() (waitFinished bool, response *iaasalpha.Image, err error) {
+		image, err := a.GetImageExecute(ctx, projectId, imageId)
+		if err != nil {
+			return false, image, err
+		}
+		if image.Id == nil || image.Status == nil {
+			return false, image, fmt.Errorf("create failed for image with id %s, the response is not valid: the id or the status are missing", imageId)
+		}
+		if *image.Id == imageId && *image.Status == ImageAvailableStatus {
+			return true, image, nil
+		}
+		if *image.Id == imageId && *image.Status == ErrorStatus {
+			return true, image, fmt.Errorf("create failed for image with id %s", imageId)
+		}
+		return false, image, nil
+	})
+	handler.SetTimeout(15 * time.Minute)
```

Comment:
the timeout is highly dependent on the size of the image, but I would say in general we can increase this to 30 minutes

Reply:
The PUT request to upload the image already blocks until the upload is complete, so I think in the normal case the waiter just waits for the state to change after the upload